src/parquet/column_reader.cc

```diff
@@ -243,4 +166,27 @@ bool ColumnReader::ReadNewPage() {
   return true;
 }
 
+ColumnReader* make_column_reader(const parquet::ColumnMetaData* metadata,
```

Comment:
It will be much safer to return std::unique_ptr<ColumnReader> instead of a naked pointer.